### PR TITLE
`Nullable<'T>.Value` → `Nullable<'T>.GetValueOrDefault`

### PR DESCRIPTION
**Pull request by brianrourkeboll**
## Description

* We are already checking `Nullable<'T>.HasValue` before accessing `Nullable<'T>.Value`, so it is safe to call `Nullable<'T>.GetValueOrDefault`. `Nullable<'T>.Value` [does a redundant check of `HasValue`](https://github.com/dotnet/runtime/blob/9fa8fd5cf77c28b465872b66745d657434706be7/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L39-L49), while `Nullable<'T>.GetValueOrDefault` [does not](https://github.com/dotnet/runtime/blob/9fa8fd5cf77c28b465872b66745d657434706be7/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L52). Using `GetValueOrDefault` is significantly faster when no value is present.

**Note**: these functions are marked `inline` — does this change have any binary compatibility implications?

## Benchmarks

| Method                    | Mean      | Error     | StdDev    | Ratio | RatioSD |
|-------------------------- |----------:|----------:|----------:|------:|--------:|
| Value_NoValue             | 0.2210 ns | 0.0140 ns | 0.0117 ns |  1.00 |    0.07 |
| GetValueOrDefault_NoValue | 0.0368 ns | 0.0106 ns | 0.0089 ns |  0.17 |    0.04 |
|                           |           |           |           |       |         |
| Value_Value               | 3.1128 ns | 0.0731 ns | 0.0610 ns |  1.00 |    0.03 |
| GetValueOrDefault_Value   | 3.0675 ns | 0.0107 ns | 0.0084 ns |  0.99 |    0.02 |

<details>
<summary>Source</summary>

```fsharp
module OfNullable.Program

open System
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Configs
open BenchmarkDotNet.Running

[<GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByCategory)>]
type Benchmarks () =
    let noValue = Nullable<int> ()
    let value = Nullable 3

    [<Benchmark(Baseline = true)>]
    [<BenchmarkCategory("NoValue")>]
    member _.Value_NoValue () = ValueOption.ofNullable noValue

    [<Benchmark>]
    [<BenchmarkCategory("NoValue")>]
    member _.GetValueOrDefault_NoValue () = if noValue.HasValue then ValueSome (noValue.GetValueOrDefault ()) else ValueNone

    [<Benchmark(Baseline = true)>]
    [<BenchmarkCategory("Value")>]
    member _.Value_Value () = ValueOption.ofNullable value

    [<Benchmark>]
    [<BenchmarkCategory("Value")>]
    member _.GetValueOrDefault_Value () = if value.HasValue then ValueSome (value.GetValueOrDefault ()) else ValueNone

ignore (BenchmarkRunner.Run<Benchmarks> ())
```

</details>

## Checklist

- [ ] Test cases added.
- [x] Performance benchmarks added in case of performance changes.
- [ ] Release notes entry updated.

### CODE REVIEW

**Comment by vzarytovskii** [comment]
That's nice, thanks. I didn't even know about this API existence.

**Review by vzarytovskii — APPROVED** [review]

**Review by vzarytovskii — APPROVED** [review]

**Review by edgarfgp — APPROVED** [review]

File: src/FSharp.Core/option.fs [hunk 1]
```diff
@@ -148,7 +148,7 @@ module Option =
     [<CompiledName("OfNullable")>]
     let inline ofNullable (value: System.Nullable<'T>) =
         if value.HasValue then
-            Some value.Value
```
Comment (psfinaki) [inline src/FSharp.Core/option.fs:151]: Hah would've assumed the opposite judging by the members' names. 
You'd do a good thing adding a comment here.

**Review by psfinaki — APPROVED** [review]
Thanks!

**Review by KevinRansom — APPROVED** [review]

File: src/FSharp.Core/option.fs (hunk 1 above)
Comment (KevinRansom) [reply src/FSharp.Core/option.fs:151]: It's fine as is.

**Review by T-Gro — APPROVED** [review]

**Pull request merged**